features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/resources/ResourcesSerializer.kt

```diff
@@ -123,33 +133,48 @@ internal class Base64Serializer private constructor(
 
         if (byteArray.isEmpty()) {
             // failed to get image data
-            base64SerializerCallback.onReady()
+            resourcesSerializerCallback.onReady()
             return
         }
 
-        val base64String = convertBitmapToBase64(
-            byteArray = byteArray,
-            bitmap = bitmap,
-            shouldCacheBitmap = shouldCacheBitmap
-        ).toByteArray(Charsets.UTF_8)
-        val resourceId = md5HashGenerator.generate(byteArray)?.toByteArray(Charsets.UTF_8)
-        val cacheData = CacheData(base64String, resourceId)
+        val resourceId = md5HashGenerator.generate(byteArray)
 
-        if (base64String.isNotEmpty() || resourceId != null) {
-            base64LRUCache.put(drawable, cacheData)
+        if (shouldCacheBitmap) {
+            bitmapPool?.put(bitmap)
         }
 
+        if (resourceId == null) {
+            // resourceId is mandatory for resource endpoint
+            resourcesSerializerCallback.onReady()
+            return
+        }
+
+        if (!resourceIdsSeen.contains(resourceId)) {
```

Comment:
shouldn't we do the check before ? Are we computing the `byteArray` for this resource for nothing if the resource was already seen ?

Reply:
We're calculating the md5 hash from the bytearray, so I'm not sure we can move this earlier in the flow